Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/DomUtils.h>
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+#include <AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.h>
+#include <AzFramework/DocumentPropertyEditor/ReflectionAdapter.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    struct ReflectionAdapterReflectionImpl : public AZ::Reflection::IReadWrite
+    {
+        ReflectionAdapter* m_adapter;
+        AdapterBuilder m_builder;
+        AZStd::unordered_map<AZ::TypeId, AZStd::string_view> m_defaultEditorMap;
+
+        void PopulateDefaultEditors()
+        {
+            m_defaultEditorMap[azrtti_typeid<bool>()] = Nodes::CheckBox::Name;
+            m_defaultEditorMap[azrtti_typeid<uint8_t>()] = Nodes::UintSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<uint16_t>()] = Nodes::UintSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<uint32_t>()] = Nodes::UintSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<uint64_t>()] = Nodes::UintSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<int8_t>()] = Nodes::IntSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<int16_t>()] = Nodes::IntSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<int32_t>()] = Nodes::IntSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<int64_t>()] = Nodes::IntSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<float>()] = Nodes::DoubleSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<double>()] = Nodes::DoubleSpinBox::Name;
+            m_defaultEditorMap[azrtti_typeid<AZStd::string>()] = Nodes::LineEdit::Name;
+        }
+
+        ReflectionAdapterReflectionImpl(ReflectionAdapter* adapter)
+            : m_adapter(adapter)
+        {
+            PopulateDefaultEditors();
+        }
+
+        AZStd::string_view GetPropertyEditor(const AZ::TypeId& id, const Reflection::IAttributes& attributes)
+        {
+            (void)attributes;
```

Comment:
This should use `[[maybe_unused]]` to mark the attributes as unused
```suggestion
        AZStd::string_view GetPropertyEditor(const AZ::TypeId& id, [[maybe_unused]] const Reflection::IAttributes& attributes)
        {
```